### PR TITLE
Bump version to 0.6.1

### LINKED ISSUE #84: There is a new release that includes the fixes for .conjurrc parsing
AC:
- [ ] Release a version of the Go client that includes the fix for #75

### PR DESCRIPTION
**Pull request by izgeri**
### What does this PR do?
- Bumps version to 0.6.1
- Also adds release instructions to CONTRIBUTING and adds a NOTICES.txt file

### What ticket does this PR close?
Resolves #84 

#### Change log
- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

### CODE REVIEW

**Review by BradleyBoutcher — APPROVED** [review]
LGTM ✨

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -25,3 +25,23 @@ Kick off your TDD (i.e. goconvey powered) development environment as follows:
...
+In particular, for each release you should:
+
+- Review the changes since the last release, and update the [version](./VERSION)
+  following [semantic versioning](https://semver.org/).
+- Determine whether any dependencies were added, removed, or updated in
+  [`go.mod`](./go.mod) since the last release. If any changes have been made,
+  update the [NOTICES](./NOTICES.txt) file.
+- Update the [CHANGELOG](./CHANGELOG.md) to reflect the new version.
+- Commit the changes to the files above in a branch and submit a version bump PR
+- Once the PR has been merged, tag the version using
```
Comment (diverdane) [inline CONTRIBUTING.md:42]: For tagging and creating a release, do we want to add a reference to the community guidelines?:
https://github.com/cyberark/community/blob/master/Conjur/CONTRIBUTING.md#tagging
https://github.com/cyberark/community/blob/master/Conjur/CONTRIBUTING.md#github-releases

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -25,3 +25,23 @@ Kick off your TDD (i.e. goconvey powered) development environment as follows:
...
+  `git tag -s vx.y.z -m vx.y.z`. Note: this requires you to be able to sign
+  releases. Consult the [github documentation on signing commits](https://help.github.com/articles/signing-commits-with-gpg/).
+- Push the tag by running `git push origin vx.y.z`
+- Create a GitHub release for the tag, and copy the changelog for this version
```
Comment (diverdane) [inline CONTRIBUTING.md:46]: Do we want to mention that the release should include a Zip and a tar.gz of the source as artifacts?

**Review by diverdane — COMMENTED** [review]
Looks good! A couple of suggestions, not sure if they're helpful or not.

File: CONTRIBUTING.md (hunk 1 above)
Comment (izgeri) [reply CONTRIBUTING.md:42]: I link to the overall release process above, which links to both of these

File: CONTRIBUTING.md (hunk 2 above)
Comment (izgeri) [reply CONTRIBUTING.md:46]: I don't think we do anything special for this, we just use what automatically gets put in the release

Comment (diverdane) [reply CONTRIBUTING.md:46]: Right! I was just checking the Github release docs, looks like that happens automagically.

**Pull request merged**